runtime/src/tokio/runtime.rs

```diff
@@ -278,6 +279,18 @@ impl crate::Runner for Runner {
     }
 }
 
+impl AsyncExecutor for &Runner {
```

Comment:
A proper implementation of this is blocking this (but should be close).

Reply:
PR has been amended to build on your new runtime changes.